WordPress/Classes/ViewRelated/Themes/ThemeBrowserViewController.swift

```diff
@@ -0,0 +1,201 @@
+            sectionNameKeyPath: nil,
+            cacheName: nil)
+        frc.delegate = self
+        
+        do {
+            try frc.performFetch()
+        } catch let error as NSError {
+            DDLogSwift.logError("Error fetching themes: \(error.localizedDescription)")
+        }
+        
+        return frc
+    }()
+    
+    /**
+     *  @brief      The themes service we'll use in this VC.
+     */
+    private lazy var themeService : ThemeService = {
+        ThemeService(managedObjectContext: self.managedObjectContext)
+    }()
+    
+    // MARK: - Additional initialization
+    
+    public func configureWithBlog(blog: Blog) {
+        do {
+            let blogInContext = try managedObjectContext.existingObjectWithID(blog.objectID) as? Blog
+            self.blog = blogInContext
+        } catch let error as NSError {
+            DDLogSwift.logError("Error finding blog: \(error.localizedDescription)")
+        }
+    }
+    
+    // MARK: - UIViewController
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        title = NSLocalizedString("Themes", comment: "Title of Themes browser page")
+        
+        WPStyleGuide.configureColorsForView(view, collectionView:collectionView)
+        
+        updateThemes()
+    }
+
+    public override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
+        super.didRotateFromInterfaceOrientation(fromInterfaceOrientation)
+        
+        collectionView?.collectionViewLayout.invalidateLayout()
+    }
+    
+    @available(iOS 8.0, *)
+    public override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+        
+        collectionView?.collectionViewLayout.invalidateLayout()
+    }
+
+    // MARK: - Updating the list of themes
+    
+    private func updateThemes() {
+        
+        if collectionView(collectionView!, numberOfItemsInSection: 0) == 0 {
+            let title = NSLocalizedString("Fetching Themes...", comment:"Text displayed while fetching themes")
+            WPNoResultsView.displayAnimatedBoxWithTitle(title, message: nil, view: self.view)
+        }
+
+        themeService.getThemesForBlog(
+            blog,
+            success: { [weak self] (themes : [AnyObject]?) -> Void in
+                WPNoResultsView.removeFromView(self?.view)
+            },
+            failure: {(error : NSError!) -> Void in
+                DDLogSwift.logError("Error updating themes: \(error.localizedDescription)")
+            })
+    }
+    
+    // MARK: - UICollectionViewController protocol UICollectionViewDataSource
+    
+    public override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let count = themesController.fetchedObjects?.count
+        return count ?? 0
```

Comment:
(Absolutely no need to change anything!) `return themesController.fetchedObjects?.count ?? 0`